recurring_contract/models/recurring_contract.py

```diff
@@ -514,17 +521,18 @@ def _updt_invoices_rc(self, vals):
 
         :param vals: the values that are being updated on the contract
         """
-        if any(key in vals for key in ("group_id", "contract_line_ids", "birthday_invoice", "christmas_invoice")):
-            data_invs = {}
-            for contract in self:
-                for inv in contract.mapped("invoice_line_ids.move_id").filtered(
-                        lambda m: m.payment_state == "not_paid" and m.state != "cancel"):
-                    data_invs.update(
-                        inv._build_invoice_data(
-                            contract=contract,
-                            ref=contract.group_id.ref,
-                            pay_mode_id=contract.group_id.payment_mode_id
-                        )
-                    )
-            if data_invs:
-                self.mapped("invoice_line_ids.move_id").update_open_invoices(data_invs)
+        data_invs = {}
+        if "contract_line_ids" in vals:
+            data_invs = self.mapped("open_invoice_ids")._build_invoice_data(contracts=self)
+        if "group_id" in vals:
```

Comment:
If we update the partner_id will it update group_id ? Is this why we don't catch partner_id there ?

Reply:
Yes, when changing the partner, the group also changes because we cannot use a group from another partner. Note that in the previous code version, the `partner_id` field was also not in the list of trigger fields here.